src/org/stringtemplate/v4/STGroup.java

```diff
@@ -745,8 +745,9 @@ public void registerModelAdaptor(Class<?> attributeType, ModelAdaptor adaptor) {
         adaptors.put(attributeType, adaptor);
     }
 
-    public ModelAdaptor getModelAdaptor(Class<?> attributeType) {
-        return adaptors.get(attributeType);
+    public <T> ModelAdaptor<T> getModelAdaptor(Class<? extends T> attributeType) {
```

Comment:
I actually missed the argument type for `getAttributeRenderer` in the previous review. It needs to also be changed to `Class<T>`.

Reply:
I'll address this once we agree on the signature of getModelAdaptor. Should I make a new PR or just add it to this one?

Reply:
It can be added to this one